### PR TITLE
✨ Add settings actions

### LINKED ISSUE #439: Add WorkspaceSelected action
## Description
Add `WorkspaceSelected` action. It should store the selected WS in the `UserPrefs` object in the state. Add the missing property if needed.

Check [notion](https://www.notion.so/toggl/Settings) for details regarding parameters and effects

## Dependencies
#435
## Definition of done
- [ ] Add `WorkspaceSelected` action
- [ ] Add tests

### LINKED ISSUE #440: Add DateFormatSelected action
## Description
Add `DateFormatSelected` action. It should store the selected WS in the `UserPrefs` object in the state. Add the missing property if needed. An enum could be a good idea for representing the possible dateFormat options.

Check [notion](https://www.notion.so/toggl/Settings) for details regarding parameters and effects

## Dependencies
#435
## Definition of done
- [ ] Add `DateFormatSelected` action
- [ ] Add needed model for representing the data
- [ ] Add tests

### LINKED ISSUE #441: Add Use24HourClockToggled action
## Description
Add `Use24HourClockToggled` action. It should update the `UserPrefs` object in the state. Add the missing property if needed.

Check [notion](https://www.notion.so/toggl/Settings) for details regarding parameters and effects

## Dependencies
#435
## Definition of done
- [ ] Add Use24HourClockToggled action
- [ ] Add tests

### LINKED ISSUE #442: Add DurationFormatSelected action
## Description
Add `DurationFormatSelected` action. It should update the `UserPrefs` object in the state. Add the missing property if needed. An Enum could a good way to keep all the possible duration formats.

Check [notion](https://www.notion.so/toggl/Settings) for details regarding parameters and effects

## Dependencies
#435
## Definition of done
- [ ] Add DurationFormatSelected action
- [ ] Add needed model for representing the data
- [ ] Add tests

### PR DESCRIPTION
**Pull request by semanticer**
### Summary
This PR adds a couple of settings actions which change UserPreferences

### Technical considerations
Nothing very interesting, mostly just boilerplate.

### Why is it done like this and not some other way?
🤷‍♂️ 

### Relationships

Closes #439 
Closes #440  
Closes #441  
Closes #442 

### Pull Request checklist

- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [x] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [x] One of our [live or file custom templates](https://github.com/toggl/android/tree/master/.idea/templates) was used in this PR

### Pointers
Should be easy enough to review at once 

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
🦑 it

### CODE REVIEW

File: settings/src/main/java/com/toggl/settings/domain/SettingsAction.kt [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 
 import arrow.optics.optics
+import com.toggl.models.domain.DateFormat
+import com.toggl.models.domain.DurationFormat
 import com.toggl.models.domain.UserPreferences
 
 @optics
 sealed class SettingsAction {
     data class UserPreferencesUpdated(val userPreferences: UserPreferences) : SettingsAction()
     data class ManualModeToggled(val isManual: Boolean) : SettingsAction()
```
Comment (daividssilverio) [inline settings/src/main/java/com/toggl/settings/domain/SettingsAction.kt:11]: Just one thing thought. Isn't this gonna cause the view to tell the value of the state?
Shouldn't we use the existing value of the state the modify it instead?

Comment (semanticer) [reply settings/src/main/java/com/toggl/settings/domain/SettingsAction.kt:11]: You are right, we are now more dependent on the state of our view, however, what you suggest would work only for boolean preferences so I would slightly prefer a current solution in the name of consistency.

Comment (daividssilverio) [reply settings/src/main/java/com/toggl/settings/domain/SettingsAction.kt:11]: Makes sense.

File: settings/src/test/java/com/toggl/settings/domain/UpdateUserPreferencesEffectTests.kt [hunk 2]
```diff
@@ -17,7 +19,13 @@ class UpdateUserPreferencesEffectTests : CoroutineTest() {
...
     @Test
     fun `Should save and return new user preferences`() = runBlockingTest {
-        val newUserPreferences = UserPreferences(isManualModeEnabled = true)
+        val newUserPreferences = createUserPreferences(
+            isManualModeEnabled = true,
+            is24HourClock = true,
+            selectedWorkspaceId = 1,
+            dateFormat = DateFormat.DDMMYYYY_dash,
+            durationFormat = DurationFormat.Decimal
+        )
```
Comment (daividssilverio) [inline settings/src/test/java/com/toggl/settings/domain/UpdateUserPreferencesEffectTests.kt:28]: I wanted to see the other lines similar to this one bellow:
```
 resultAction.userPreferences.isManualModeEnabled.shouldBeTrue()
```

**Pull request merged**